### PR TITLE
90-query-for-cities

### PR DESCRIPTION
**Pull request by Avraham-S**
moved functions to designated folder

### CODE REVIEW

File: firebase/users.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { getAuth } from "firebase/auth";
+import { db } from "../firebaseConfig";
+import { doc, getDoc, updateDoc, arrayUnion } from "firebase/firestore";
+
+const fetchUsersCities = async (setStateFunction) => {
```
Comment (softwarebyze) [inline firebase/users.js:5]: Any reason why we can't just return the cities?

File: firebase/users.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const auth = getAuth();
+    const userId = auth.currentUser?.uid;
+    const userRef = doc(db, "users", userId);
+    const userSnap = await getDoc(userRef);
+    if (userSnap.exists()) {
+      setStateFunction(() => {
+        const cities = userSnap.data().cities || [userSnap.data().location];
+        return cities.map((city, i) => ({
+          city: city,
+          id: Date.now() + i + Math.random(),
```
Comment (softwarebyze) [inline firebase/users.js:16]: Why do we need to add an id here?

**Review by softwarebyze — CHANGES_REQUESTED** [review]

File: firebase/users.js (hunk 1 above)
Comment (Avraham-S) [reply firebase/users.js:5]: We could do that. I'll change it

File: firebase/users.js (hunk 2 above)
Comment (Avraham-S) [reply firebase/users.js:16]: It was for the react key, but I'll remove it and just use the city name

File: components/pages/HomePage.js [hunk 3]
```diff
@@ -137,25 +124,9 @@ const HomePage = () => {
...
-          {},
-          { limit: 30 },
-        );
-        setChannelList(channels);
-      } catch (error) {
-        console.error(error);
-      }
-    };
-    fetchChannels();
+    fetchUsersCities().then((cities) => setCities(cities));
```
Comment (softwarebyze) [inline components/pages/HomePage.js:127]: i prefer async await syntax here (and creating a function inside the useEffect)

File: firebase/users.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      return cities;
+    } else {
+      throw new Error("No user found");
+    }
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const addUserCity = async (city, usersCities) => {
```
Comment (softwarebyze) [inline firebase/users.js:22]: id rather just take the one city to add here and fetch the current cities within the function

**Review by softwarebyze — CHANGES_REQUESTED** [review]
After these changes, can you please add a demo screen recording

**Comment by Avraham-S** [comment]
https://github.com/softwarebyze/Places/assets/103695563/45036b31-3ec4-4d1f-b2e7-16a29388e12b

No new functionality, just refactoring

**Review by softwarebyze — APPROVED** [review]

**Pull request merged**